sw/device/lib/dif/dif_gpio.c

```diff
@@ -70,96 +70,112 @@ void (*abort_if_invalid_fptr)(dif_gpio_t const *const) = abort_if_invalid;
  *
  * @param index Bit index in a 32-bit register.
  */
-static inline uint32_t index_to_mask(uint32_t const index) {
+static inline uint32_t index_to_mask(const uint32_t index) {
   if (index >= (8 * sizeof(uint32_t))) {
     abort();
   }
   return 1U << index;
 }
 #ifdef GTEST_TEST
-uint32_t (*index_to_mask_fptr)(uint32_t const) = index_to_mask;
+uint32_t (*index_to_mask_fptr)(const uint32_t) = index_to_mask;
 #define index_to_mask index_to_mask_fptr
 #endif
 
-static inline uint32_t read_register(dif_gpio_t const *const gpio,
-                                     uint32_t const offset) {
+static inline uint32_t read_register(const dif_gpio_t *const gpio,
+                                     const uint32_t offset) {
   abort_if_invalid(gpio);
   return reg32_read(gpio->base_addr, offset);
 }
 #ifdef GTEST_TEST
-uint32_t (*read_register_fptr)(dif_gpio_t const *const gpio,
-                               uint32_t const offset) = read_register;
+uint32_t (*read_register_fptr)(const dif_gpio_t *const gpio,
+                               const uint32_t offset) = read_register;
 #define read_register read_register_fptr
 #endif
 
-static inline bool read_register_bit(dif_gpio_t const *const gpio,
-                                     uint32_t const offset,
-                                     uint32_t const index) {
+static inline bool read_register_bit(const dif_gpio_t *const gpio,
+                                     const uint32_t offset,
+                                     const uint32_t index) {
   return read_register(gpio, offset) & index_to_mask(index);
 }
 #ifdef GTEST_TEST
-bool (*read_register_bit_fptr)(dif_gpio_t const *const gpio,
-                               uint32_t const offset,
-                               uint32_t const index) = read_register_bit;
+bool (*read_register_bit_fptr)(const dif_gpio_t *const gpio,
+                               const uint32_t offset,
+                               const uint32_t index) = read_register_bit;
 #define read_register_bit read_register_bit_fptr
 #endif
 
-static inline void write_register(dif_gpio_t const *const gpio,
-                                  uint32_t const offset, uint32_t const val) {
+static inline void write_register(const dif_gpio_t *const gpio,
```

Comment:
`read_register` and `write_register` were implemented to guard against null pointers, but since we don't do that anymore, I will remove these functions.